src/middlewares/validator.js

```diff
@@ -1,13 +1,40 @@
 const createError = require('http-errors')
 const Ajv = require('ajv')
 const ajvKeywords = require('ajv-keywords')
+const ajvLocalize = require('ajv-i18n')
 const {deepEqual} = require('assert')
 
 let ajv
 let previousConstructorOptions
-const defaults = {v5: true, $data: true, allErrors: true}
+const defaults = {
+  v5: true,
+  coerceTypes: 'array', // important for query string params
+  allErrors: true,
+  useDefaults: true,
+  $data: true // required for ajv-keywords
+}
+
+module.exports = ({ inputSchema, outputSchema, ajvOptions, errorFormat = errors => errors }) => {
```

Comment:
On one side, I like the idea of the `errorFormat` function. On the other, I would prefer to keep this middleware simpler and encourage people to combine it with the usage of the [HTTP Error Handler middleware](https://github.com/middyjs/middy/blob/master/docs/middlewares.md#httperrorhandler).

Reply:
I've been writing a `jsonapi` middleware that essentially replaces `httpErrorHandler`. At the time of writing it made sense to do the formatting here, there have been changes since that might make more sense to move it to another middleware. Let me get back to you on this one.